responder/events.go

```diff
@@ -48,3 +50,43 @@ func NotifyJiraResponder(responder api.Responder) (string, error) {
 
 	return issue.Key, nil
 }
+
+func NotifyMSPlannerResponder(responder api.Responder) (string, error) {
+	if responder.Properties["responderType"] != "MSPlanner" {
+		return "", fmt.Errorf("invalid responderType: %s", responder.Properties["responderType"])
+	}
+
+	teamSpecJson, err := responder.Team.Spec.MarshalJSON()
+	if err != nil {
+		return "", err
+	}
+
+	var teamSpec api.TeamSpec
+	if err = json.Unmarshal(teamSpecJson, &teamSpec); err != nil {
+		return "", err
+	}
+
+	client, err := msplanner.NewClient(
+		teamSpec.ResponderClients.MSPlanner.TenantID,
+		teamSpec.ResponderClients.MSPlanner.ClientID,
+		teamSpec.ResponderClients.MSPlanner.GroupID,
+		teamSpec.ResponderClients.MSPlanner.Username.Value,
+		teamSpec.ResponderClients.MSPlanner.Password.Value,
```

Comment:
Will we be getting the value directly or do we need to first fetch it from a secret?
Like we have in canary-checker at `checks/common.go:L65`

Reply:
We'll be fetching the secret and storing it in the database, and the database would be the source of truth. So this flow assumes that the k8s object -> database part is already done.